pennylane/templates/broadcast.py

```diff
@@ -223,21 +267,43 @@ def circuit(hadamard=None):
         "Iterable; got {}".format(type(parameters)),
     )
 
+    _check_type(
+        structure, [str], msg="'structure' must be a string; got {}".format(type(structure)),
+    )
+
+    _check_is_in_options(
+        structure, OPTIONS, msg="did not recognize option {} for 'structure'".format(structure),
+    )
+
+    n_parameters = {
+        "single": len(wires),
+        "double": len(wires) // 2,
+        "double_odd": (len(wires) - 1) // 2,
+    }
+
+    # check that enough parameters for structure
     if parameters is not None:
         shape = _get_shape(parameters)
-        if shape[0] != len(wires):
+        if shape[0] != n_parameters[structure]:
             raise ValueError(
-                "'parameters' must contain one entry for each of the {} wires; got shape {}".format(
-                    len(wires), shape
+                "'parameters' must contain entries for {} blocks; got {} entries".format(
+                    n_parameters[structure], shape[0]
                 )
             )
         # repackage for consistent unpacking
         if len(shape) == 1:
             parameters = [[p] for p in parameters]
     else:
-        parameters = [[] for _ in range(len(wires))]
+        parameters = [[] for _ in range(n_parameters[structure])]
 
     #########
 
-    for w, p in zip(wires, parameters):
+    wire_sequence = {
+        "single": wires,
```

Comment:
Here I still would be in favour for having some sort of check for making sure that the user did not pass the same wire multiple times in `wires`.

An example is as follows, where due to the second `CNOT` being applied, the result changes:
```python
@qml.qnode(dev)  
def circuit():
    qml.Hadamard(0)
    qml.broadcast(qml.CNOT, wires=[1,0,0,1], structure='double')
    return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
```
This might be something the user doesn't do on purpose (maybe wires is just being passed on).
So a check like
```python
if len(set(wires)) != len(wires):
    #error: some wires were repeated
```
or the like could be included.

Reply:
It would be also worth having a test case for this

Reply:
This should be added to ALL templates, which can be done easily - just add to the ``_check_wires`` test. I will make a separate PR if you don't mind, since it applies to all templates.

Reply:
Sure, that sounds really good, thanks! :)